lib/runtime-c-api/src/metering.rs

```diff
@@ -0,0 +1,111 @@
+use crate::{
+    error::update_last_error, instance::wasmer_instance_t, module::wasmer_module_t, wasmer_result_t,
+};
+use std::slice;
+
+#[cfg(feature = "metering")]
+use wasmer_runtime_core::backend::Compiler;
+
+/// Creates a new Module with gas limit from the given wasm bytes.
+///
+/// Returns `wasmer_result_t::WASMER_OK` upon success.
+///
+/// Returns `wasmer_result_t::WASMER_ERROR` upon failure. Use `wasmer_last_error_length`
+/// and `wasmer_last_error_message` to get an error message.
+#[allow(clippy::cast_ptr_alignment)]
+#[cfg(feature = "metering")]
+#[no_mangle]
+pub unsafe extern "C" fn wasmer_compile_with_limit(
```

Comment:
I would rename the function `wasmer_compile_with_metering` or `_with_gas_metering` for being more explicit.

Reply:
Yes, better name